### DISCUSSION TITLE
Finally, there is a startup need talk :)

### DISCUSSION THREAD

**Discussion by SettingDust** [extractus/extractus#1] [General]
As continue of https://github.com/orgs/extractus/discussions/2
I write a simple prototype @ndaidong

**Comment by ndaidong** [comment]
@SettingDust hello, do you have any new ideas? I'm woking on few projects and almost no time to think about this.
I will take a look at the code. Could you add the install guide?

**Reply by SettingDust** [reply]
README added
Just implement the Input Html -> Extractors -> Transformer -> Selector -> Result

**Comment by ndaidong** [comment]
@SettingDust great job! Thank you so much. I will test it and give you the feedback soon.

**Comment by ndaidong** [comment]
@SettingDust I cloned the repo. then run the following command:

```
pnpm i
pnpm test
```

Then I got some errors:

![Screenshot from 2023-06-03 06-58-49](https://github.com/extractus/workspace/assets/1809619/2d379cc9-68c1-4481-874f-c9526d722a46)

![Screenshot from 2023-06-03 06-55-51](https://github.com/extractus/workspace/assets/1809619/029ba0c4-83cc-4baf-bba6-97f18928b155)

Is this normal or I'm missing something else?

**Reply by SettingDust** [reply]
Noop. I'll provide fix ASAP. It's fine before orz. Can be treat as normal and just run test for `package/extractus/index.spec.ts`

**Reply by SettingDust** [reply]
Add `DEBUG=true` to env ")

**Reply by ndaidong** [reply]
`DEBUG=true pmpm test`? it gives the same output.

**Reply by ndaidong** [reply]
otherwise, the architecture looks cool, I start following your deisgn idea.

**Reply by SettingDust** [reply]
> `DEBUG=true pmpm test`? it gives the same output.

`DEBUG=true pmpm test`, but for packages/extractus. There is a task in the package

**Reply by ndaidong** [reply]
ah I see, thanks

**Reply by ndaidong** [reply]
I've `pnpm i` from root folder and packages/extractus, but it still misses the memorize:

![Uploading Screenshot from 2023-06-03 07-36-49.png…]()

**Reply by SettingDust** [reply]
That's my fault. The file not import memoize. I've pushed a new commit. You can do that when you taking time.

**Reply by ndaidong** [reply]
@SettingDust I've removed the whole repo and clone/install again. Now I faced another error:
![Screenshot from 2023-06-04 10-38-12](https://github.com/extractus/workspace/assets/1809619/39326b10-b6bf-4291-b07a-a65c1a10d444)

**Reply by SettingDust** [reply]
`DEBUG=true` for now. The deep merge not works well on iterable now :(.

**Reply by ndaidong** [reply]
@SettingDust still error :D

Does it pass all tests on your machine? If so, you should check the dev environment again.

![Screenshot from 2023-06-04 13-29-22](https://github.com/extractus/workspace/assets/1809619/7126ad3d-89de-4c1d-9b91-da9936a88649)

**Reply by SettingDust** [reply]
Error is expected. The test of extract haven't fill with expected value that the extractors and deep merge isn't complete.
Just for show the prototype

**Reply by SettingDust** [reply]
As you can see. The deep equal expect `{}` that isn't a valid output

**Reply by ndaidong** [reply]
@SettingDust I thought if it was an error, it would have to be caught.
OK, so what is the next plan? Should we release the first RC version?

**Reply by SettingDust** [reply]
The error that can't run without DEBUG should be fixed.
I complete this for your reviewing. If you think it's fine. Then I need add more extractors, transfprmer, selector as I did in my repo

**Reply by ndaidong** [reply]
got it, thank you.

**Comment by ndaidong** [comment]
@SettingDust what is your [npmjs.com](https://www.npmjs.com/) usename? I would like to add you to the organization Extractus so you can manage the packages by your self.

**Reply by SettingDust** [reply]
https://www.npmjs.com/~settingdust

**Reply by ndaidong** [reply]
@SettingDust I've sent invitation to join organization, please check if you received

**Reply by SettingDust** [reply]
joined

**Reply by ndaidong** [reply]
yeah, so you can update and publish packages by your self. It's good to start with some RC version for test. 
The overall structure of the modules in this repo, I found to be well organized.
